### PR TITLE
Rename various modules and functions

### LINKED ISSUE #43: rename or move many objects/files
`ConfigSet_in` -> `ConfigSet`, `ConfigSet_out` -> `OutputSpec`
`pipeline/` -> `autoparallelize/`, `iterable_loop` -> `autoparallelize
All low level `*_op` -> `*_autopara_wrappable`
`generate_configs/` -> `generate/`
`select_configs/` -> `select/`

move `selection_space.py` -> `select/`
move `mpipool_support.py` -> `autoparallelize`
move `calc_descriptors.py` -> `descriptors/calc.py`

### PR DESCRIPTION
**Pull request by bernstei**
More sensible names and organization for some important (and less important) modules and functions

closes #43

### CODE REVIEW

**Comment by bernstei** [comment]
@gabor1 @gelzinyte who should review this?

**Comment by gelzinyte** [comment]
I'm looking through this

File: wfl/autoparallelize/pool.py
Comment (gelzinyte) [inline wfl/autoparallelize/pool.py:108]: Line 112 often translates to 
```
 Running <function evaluate_op at 0x145549c97160> with npool=16, chunksize=1
```
which is not perfectly helpful, because most of the `op`'s names aren't very helpful. Could this also report the module of the function? Or alternatively, should we give better names to all of `*autopara_wrappable' functions?

Comment (bernstei) [reply wfl/autoparallelize/pool.py:108]: I agree that it's not helpful.  I wasn't thinking that they needed to have a sensible name, but because of this sort of thing they do. Maybe do that as part of the `autoparallelize` manual wrap to `iloop` that's in issue #48 ?

File: wfl/descriptors/calc.py
Comment (gelzinyte) [inline wfl/descriptors/calc.py:8]: Why not to rename this to `wfl/descriptors.py` or something similar? Especially because there is a `calc` function there already??

File: wfl/autoparallelize/pool.py
Comment (gelzinyte) [reply wfl/autoparallelize/pool.py:108]: Sounds good, I've just made a note there.

**Comment by bernstei** [comment]
also closes #45

File: wfl/descriptors/calc.py
Comment (gelzinyte) [inline wfl/descriptors/calc.py:76]: Is it worth renaming `calc` to something more descriptive? I am thinking of using it in code - I would be inclined to rename it when importing (like in `devtools/scripts/pretty_pca.py`) so it doesn't clash with other `calc` methods (if there are any) and the code using it is more readable.

File: wfl/descriptors/calc.py
Comment (bernstei) [reply wfl/descriptors/calc.py:8]: Now I'm confused - this used to be `wfl/descriptors.py`  @gabor1 do you remember why we said we should move it into a subdirectory, given that there isn't anything else in `wfl/descriptors/`?

File: wfl/calculators/dft.py
Comment (gelzinyte) [inline wfl/calculators/dft.py:85]: Do you think `wdir` might be a bit more self-explanatory/common than `rundir`? I think `base` bit could be slightly more self-explanatory too, would `root` be any better? So `wdir_root`?

Comment (bernstei) [reply wfl/calculators/dft.py:85]: I don't find `wdir` to be any better - maybe `workdir`?

File: wfl/descriptors/calc.py
Comment (bernstei) [reply wfl/descriptors/calc.py:76]: Can you do `from wfl import descriptors` and then use it as `descriptors.calc()`?  Anyway, let's wait until @gabor1 reminds me why we decided to move it into a subdirectory.

File: wfl/configset.py
Comment (gelzinyte) [inline wfl/configset.py:329]: I find the `all_or_none=True, force=True` combination for skipping operations that would overwrite the `OutputSpec` files a bit cryptic. Would it be worth adding a tag `skip_if_complete` (or similar) in addition or instead?

File: wfl/descriptors/calc.py
Comment (gelzinyte) [reply wfl/descriptors/calc.py:76]: Yes, that would work perfectly fine.

File: wfl/configset.py
Comment (bernstei) [reply wfl/configset.py:329]: The problem is that the actual skipping functionality is in `autoparallelize`, not really a part of the output spec.   It would definitely be slightly more sensible if `force=True` wasn't necessary for the skipping.  

Anyway, I agree that putting those in is a bit obscure, but isn't that the default behavior now anyway, so users don't really have to worry about it?

Comment (bernstei) [reply wfl/configset.py:329]: OK.  I stand by my earlier statement - the actual skipping functionality is in `autoparallelize()`, and it just requires an `OutputSpec` setting that makes it safe.  I also stand (mostly) by my initial claim that the `OutputSpec` arguments communicate what it's going to do, but it's just that it's not obvious that those behaviors are required to get safe skipping behavior.  The "mostly" is because I do agree that it'd be better if `force` didn't matter for the skipping.  I might be able to fix that, if we think it's worth it.

However, for what it's worth, you **do not** have to pass force=True.  If you pass `all_or_none=True` (which is actually the default now) and **no** `force`, it defaults to `force=True`, and the skipping works.

File: wfl/descriptors/calc.py
Comment (bernstei) [reply wfl/descriptors/calc.py]: renamed to `descriptors/quippy.py`

File: wfl/descriptors/calc.py
Comment (bernstei) [reply wfl/descriptors/calc.py]: It's now `descriptors.quippy.calc`. If that's OK, do you want to resolve this conversation?

File: wfl/calculators/dft.py
Comment (bernstei) [reply wfl/calculators/dft.py:85]: Are you going to do that, or should I?

Comment (gabor1) [reply wfl/calculators/dft.py:85]: I would prefer workdir

**Review by gabor1 — APPROVED** [review]
I scrolled through all the renamings and all looked good to me.

File: wfl/calculators/dft.py
Comment (gelzinyte) [reply wfl/calculators/dft.py:85]: > Are you going to do that, or should I?
I'll do that in a bit

File: wfl/configset.py
Comment (gelzinyte) [reply wfl/configset.py:329]: okay! I like that `all_or_none=True` (the default) sets the `force` as well. 

I think "all_or_none" came from slightly different reasoning than "skip if the file is present", but I think that's going to be the more frequent use./? What do you think about renaming "all_or_none" to reflect that and be slightly more self-explanatory?

File: wfl/generate/minim.py
Comment (gelzinyte) [inline wfl/generate/minim.py:7]: Do you think `optimize.py` (following ASE) would be ever so slightly clearer than `minim.py`?

File: wfl/generate/vib.py
Comment (gelzinyte) [inline wfl/generate/vib.py:14]: let's rename this to `normal_modes.py`?

File: wfl/autoparallelize/base.py
Comment (gelzinyte) [inline wfl/autoparallelize/base.py:89]: Did we talk about looking for clearer names for `chunksize` and `job_chunksize`? 

not sure if any of these are any better for `chunksize`: `serial_chunk_size`/`serial_chunksize`/`num_serial`/`num_serial_items`/`num_serial_ops`/`num_operations_in_serial`

instead of `job_chunksize` maybe `items_per_job`?

File: wfl/fit/gap_simple.py
Comment (gelzinyte) [inline wfl/fit/gap_simple.py:4]: how about making a `gap` submodule? Then 
* `gap_simple.py` -> `gap/gap.py` or just in `gap/__init__.py` 
* `gap_multistage.py` -> `gap/multistage.py`
* `glue_2b.py` -> `gap/spline_2b.py` or `gap/glue.py` or `gap/glue_2b.py`
* `gap_relocate.py` -> `gap/relocate.py`

File: wfl/configset.py
Comment (bernstei) [reply wfl/configset.py:329]: If we can come up with a better name, great, but since `OutputSpec` is not the same as `autoparallelize`, and it's `autoparallelize` that does the skipping, I don't think the "skipping" part makes sense to be in the `OutputSpec` arguments.  The standard name for this behavior (an operation that's all or none) is an atomic operation, so a more technically correct (if not necessarily more clear) name for that argument could be `atomic_write`, but I'm not sure that really helps you.   The fundamental problem is that there's nothing inherent to the `OutputSpec` that makes it skip - it's only `autoparallelize` that uses certain `OutputSpec` _behavior_.

File: wfl/generate/minim.py
Comment (bernstei) [reply wfl/generate/minim.py:7]: For the sake of compatibility, I suppose that's fine.

File: wfl/autoparallelize/base.py
Comment (bernstei) [reply wfl/autoparallelize/base.py:89]: Yes, we did, but I think that could also be a different PR. We need to finish the workflow presentation/discussion,

File: wfl/autoparallelize/utils.py
Comment (bernstei) [reply wfl/autoparallelize/utils.py:14]: Yes, please.

File: wfl/fit/gap_simple.py
Comment (bernstei) [reply wfl/fit/gap_simple.py:4]: Sounds good, although I don't like only gap simple in `__init__`.

File: wfl/configset.py
Comment (gelzinyte) [reply wfl/configset.py:329]: okay, I don't have any better ideas, let's leave it as is.

File: wfl/fit/gap_simple.py
Comment (gelzinyte) [reply wfl/fit/gap_simple.py:4]: you don't like putting things in `__init__.py` or would you like to have fewer/more things in there than just `gap_simple.py` (if so, what?)?

**Comment by gelzinyte** [comment]
should I make the changes we have agreed to?

**Comment by bernstei** [comment]
> should I make the changes we have agreed to?

If you can do it, that'd be great.

File: wfl/fit/gap_simple.py
Comment (bernstei) [reply wfl/fit/gap_simple.py:4]: I don't like putting only gap_simple there.  I'd say put `gap_simple` in `simple.py`, and could go either way on having an empty `__init__` or having references to both simple and multistage.

**Comment by gelzinyte** [comment]
I'm good with the changes once we resolve the remaining conversations, thanks!

File: wfl/generate/vib.py
Comment (bernstei) [reply wfl/generate/vib.py:14]: Sure.  I'll do that.

Comment (bernstei) [reply wfl/generate/vib.py:14]: Actually, I'm thinking this isn't worth doing here. There will be probably be changes from the deprecation issue, and we need to think about how this should coexist with the (phonopy-based?) version that's applicable to periodic systems.

File: wfl/fit/gap_simple.py
Comment (bernstei) [reply wfl/fit/gap_simple.py:4]: I guess I'll do this.

**Comment by bernstei** [comment]
I just realized that not everything in what is now called `remoteinfo.py` is actually about `RemoteInfo`  - I forgot about `grouper()`. Maybe we should go back to a more generic name, or split off just the definition of `RemoteInfo`?

File: wfl/fit/gap_simple.py
Comment (bernstei) [reply wfl/fit/gap_simple.py:4]: I think this is done. @gelzinyte can you resolve this conversation if you think it's OK?

**Comment by bernstei** [comment]
As far as I know everything should be done.  Hopefully the tests will pass in the CI (locally they seemed OK).

File: wfl/fit/gap_simple.py
Comment (gelzinyte) [reply wfl/fit/gap_simple.py:4]: looks good!

**Comment by bernstei** [comment]
OK - merged the orca test PR, and will merge once all tests pass.

**Comment by bernstei** [comment]
I had to tweak some tests that run locally but on in github CI, but assuming it passes again, I'll merge.

**Pull request merged**
